tests/integration/test_workflow_yaml_validation.py

```
@@ -0,0 +1,500 @@
+            'windows-latest', 'windows-2022', 'windows-2019',
+            'macos-latest', 'macos-12', 'macos-11'
+        }
+        
+        for job_name, job_config in jobs.items():
+            runs_on = job_config.get('runs-on')
+            
+            if isinstance(runs_on, str):
+                # Allow self-hosted or standard runners
+                if runs_on not in valid_runners:
+                    assert runs_on.startswith('self-hosted') or \
+                           runs_on.startswith('${{'), \
```

Ruff (0.14.6)

Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)